core/src/main/java/org/elasticsearch/search/rescore/RescoreBuilder.java

```diff
@@ -112,27 +199,85 @@ public QueryRescorer setRescoreQueryWeight(float rescoreQueryWeight) {
         }
 
         /**
-         * Sets the original query score mode. The default is <tt>total</tt>
+         * Gets the original query weight for rescoring. The default is <tt>1.0</tt>
+         */
+        public float getRescoreQueryWeight() {
+            return this.rescoreQueryWeight;
+        }
+
+        /**
+         * Sets the original query score mode. The default is {@link QueryRescoreMode#Total}.
+         * @deprecated use {@link RescoreBuilder.QueryRescorer#setScoreMode(QueryRescoreMode)}
          */
+        @Deprecated
         public QueryRescorer setScoreMode(String scoreMode) {
```

Comment:
I think we should remove this method and insist on using the setScoreMode(QueryRescoreMode) method instead. This will make the API cleaner as only the permitted values (the ones in the enum) can be used. The parser can then use the fromString() method to convert REST request values to the enum but Java API users will have the safety of the enum